x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -48,18 +53,37 @@ export class RelatedDashboardsClient {
         `Alert with id ${this.alertId} not found. Could not fetch related dashboards.`
       );
     }
+    const [suggestedDashboards, linkedDashboards] = await Promise.all([
+      this.fetchSuggestedDashboards(),
+      this.getLinkedDashboards(),
+    ]);
+    const filteredSuggestedDashboards = suggestedDashboards.filter(
+      (suggested) => !linkedDashboards.some((linked) => linked.id === suggested.id)
+    );
+    return {
+      suggestedDashboards: filteredSuggestedDashboards,
+      linkedDashboards,
+    };
+  }
+
+  async fetchSuggestedDashboards(): Promise<SuggestedDashboard[]> {
+    const allSuggestedDashboards = new Set<SuggestedDashboard>();
+    const relevantDashboardsById = new Map<string, SuggestedDashboard>();
     const index = await this.getRuleQueryIndex();
+    if (!this.alert) {
+      throw new Error('Alert not found. Could not fetch suggested dashboards.');
+    }
     const allRelevantFields = this.alert.getAllRelevantFields();
 
     if (index) {
       const { dashboards } = this.getDashboardsByIndex(index);
-      dashboards.forEach((dashboard) => allRelatedDashboards.add(dashboard));
+      dashboards.forEach((dashboard) => allSuggestedDashboards.add(dashboard));
     }
     if (allRelevantFields.length > 0) {
-      const { dashboards } = this.getDashboardsByField(allRelevantFields);
-      dashboards.forEach((dashboard) => allRelatedDashboards.add(dashboard));
+      const { dashboards } = this.getDashboardsByField(Array.from(allRelevantFields));
```

Comment:
```suggestion
      const { dashboards } = this.getDashboardsByField(allRelevantFields);
```

I think we can cut this as `allRelevantFields` is already typed as `string[]`.